dashboard/db/migrate/20160505010203_migrate_net_promoter_score2015_data.rb

```diff
@@ -0,0 +1,39 @@
+# A migration that transfers the net promoter score survey results in the users
+# table to the survey_results table. Does NOT remove the data from the users
+# table.
+class MigrateNetPromoterScore2015Data < ActiveRecord::Migration
+  def change
+    reversible do |direction|
+      # In the up direction, we create SurveyResult records for users having
+      # a survey2015_value or survey2015_comment. For safety, we do not remove
+      # the information from the users row in this migration.
+      direction.up do
+        ActiveRecord::Base.record_timestamps = false
+        begin
+          User.where("properties LIKE '%survey2015%'").find_each do |user|
```

Comment:
I think you should do the where first since it reduces the amount of data
sent from the db by orders of magnitude (I'm not sure if the other way even
works). You can always try running locally and looking at the generated
queries first (the SQL queries sent to the db are solely determined by
whats written in ruby and independent of size of the table and other query
plan stuff so should be the same locally and in prod).
On May 5, 2016 7:32 AM, "ashercodeorg" <notifications@github.com> wrote:

In
dashboard/db/migrate/20160505010203_migrate_net_promoter_score2015_data.rb
<https://github.com/code-dot-org/code-dot-org/pull/8171#discussion_r62195029>
:

> @@ -0,0 +1,39 @@
> +# A migration that transfers the net promoter score survey results in the users
> +# table to the survey_results table. Does NOT remove the data from the users
> +# table.
> +class MigrateNetPromoterScore2015Data < ActiveRecord::Migration
> +  def change
> +    reversible do |direction|
> +      # In the up direction, we create SurveyResult records for users having
> +      # a survey2015_value or survey2015_comment. For safety, we do not remove
> +      # the information from the users row in this migration.
> +      direction.up do
> +        ActiveRecord::Base.record_
timestamps = false
> +        begin
> +          User.where("properties LIKE '%survey2015%'").find_each do |user|

Is it more performant to do User.where.find_each... or
User.find_each.where...?

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/code-dot-org/code-dot-org/pull/8171/files/f33a1393307343cb394d2a33382243c74a8f29fa#r62195029>